Kamila/add friends button

kamilababayeva opened the pull request:
This PR contains a first basic version of adding friends.

Changes:
- "Add" button in user search to add user as a friend
- Logic to store user as a friend  of a current user in firebase
- 
<img width="338" alt="image" src="https://user-images.githubusercontent.com/51053878/164977412-e6e443a9-ce02-4de2-9f71-614ea07acf9d.png">


TODO:
- change button interface when clicked to add
- do not display "Add" button if user already a friend

MaximeZmt commented on app/src/main/java/ch/sdp/vibester/activity/SearchUserActivity.kt:
```diff
@@ -1,33 +1,29 @@
     private var recyclerView: RecyclerView? = null
     private var searchEditText: EditText? = null
-    private val dbRef: DatabaseReference = Database.get().getReference("users")
+    var usersRepo= UsersRepo()
```
```suggestion
    var usersRepo = UsersRepo()
```

MaximeZmt reviewed (commented):
Minor typo found, otherwise good job 👍

jiabaow commented on app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt:
```diff
@@ -50,17 +64,48 @@ class UsersRepo @Inject constructor() {
+     * @param searchInput search text inputed by user
+     * @param callback function to call with found users by username
+     *
+     * Comment about \uf8ff:
```
I suggest putting this comment right below line 89 so that when people read the code, it's easier for them to find the explanation.

kamilababayeva replied:
done!

jiabaow commented on app/src/main/java/ch/sdp/vibester/user/User.kt:
```diff
@@ -9,5 +9,6 @@ data class User(
-) : Serializable {}
\ No newline at end of file
+    var ranking: Int = 0,
+    var uid: String = ""
```
why do we need an uid if the handle is already unique for each user?

kamilababayeva replied:
I was not sure whether the handle is correct. I saved the uid since the user is stored by it. We can discuss it next meeting.

jiabaow commented on app/src/androidTest/java/ch/sdp/vibester/activity/DownloadActivityTest.kt:
```diff
@@ -56,21 +57,21 @@ class DownloadActivityTest {
-        val extract = File("storage/emulated/0/Download", "extract_of_$songName")
-        assert(!extract.exists())
-    }
+//    @Test
```
maybe explain in a comment why this test is commented

kamilababayeva replied:
done!

jiabaow reviewed (commented):
Thanks for the improvements!
I added a few comments and one first issue which should not be forgot: 
Each time the build fails on e: /tmp/cirrus-ci-build/app/src/androidTest/java/ch/sdp/vibester/activity/SearchUserActivityTest.kt: (20, 32): Unresolved reference: UserProfile
e: /tmp/cirrus-ci-build/app/src/androidTest/java/ch/sdp/vibester/activity/SearchUserActivityTest.kt: (21, 32): Unresolved reference: UserProfileAdapter

kamilababayeva commented:
> Thanks for the improvements! I added a few comments and one first issue which should not be forgot: Each time the build fails on e: /tmp/cirrus-ci-build/app/src/androidTest/java/ch/sdp/vibester/activity/SearchUserActivityTest.kt: (20, 32): Unresolved reference: UserProfile e: /tmp/cirrus-ci-build/app/src/androidTest/java/ch/sdp/vibester/activity/SearchUserActivityTest.kt: (21, 32): Unresolved reference: UserProfileAdapter

thanks! fixed the issue with the build!

MaximeZmt reviewed (approved):
👍 Nice work !

jiabaow reviewed (approved):
LGTM
